src/ethereum/cancun/vm/interpreter.py

```diff
@@ -257,6 +261,11 @@ def execute_code(message: Message, env: Environment) -> Evm:
     code = message.code
     valid_jump_destinations = get_valid_jump_destinations(code)
 
+    if message.parent_evm and message.parent_evm.transient_storage:
```

Comment:
When would the parent not have a transient storage? Are there cases where an `Evm` would have a different `TransientStorage` than all of its parents?

Why not put the `TransientStorage` on the `Environment` instead?

Reply:
Updated the if condition. The `TransientStorage` was moved from the `Environment` to `Evm` data class after [this](https://github.com/ethereum/execution-specs/pull/850#pullrequestreview-1706392390) discussion.